Fix issue 4135

soulshadow opened the pull request:
This change addresses issue #4135 

Originally on line number 1467 the address of the original queue was mistakenly assigned instead of the duplicate created a few lines earlier, thereby allowing the existing queue to be modified.

In function save_queue_file()
Change:
         queueDict = ghb_array_get(ud->queue, ii);

to:
        queueDict = ghb_array_get(queue, ii);

and reorder operations to avoid unnecessary work should the file selection operation be canceled by the user. 

Also change the file GTK dialog title in save_queue_file() to "Export Queue" and in open_queue_file() to "Import Queue" to better align with the naming conventions used for other dialog windows in the GUI (Both functions originally used the title "Queue Destination")


**Before Posting:**

- [x] Create an issue describing the change. If an issue already exists, please use this.
- [x] Discuss the issue with the HandBrake team to ensure that the idea has been accepted. (An open enhancement request does not equal acceptance). This will avoid any time wasted on features that are outside the project scope!


**Description of Change:**





**Test on:**

- [ ] Windows 10+  (via MinGW)
- [ ] macOS 10.13+
- [ ] Ubuntu Linux

**Screenshots (If relevant):**


**Log file output (If relevant):**

soulshadow committed:
Fix issue 4135

This change addresses issue #4135 

Original line #1467 mistakenly assigned the address of the original queue instead of the duplicate create a few lines earlier, thereby allowing the existing queue to be modified.

In function save_queue_file()
Change:
         queueDict = ghb_array_get(ud->queue, ii);

to:
        queueDict = ghb_array_get(queue, ii);

and reorder operations to avoid unnecessary work should the file selection operation be canceled by the user. 

Also change the file GTK dialog title in save_queue_file() to "Export Queue" and in open_queue_file() to "Import Queue" to better align with the naming conventions used for other dialog windows in the GUI (Both functions originally used the title "Queue Destination")

sr55 commented:
Thanks for contributing!  

Ping @jstebbins when you have a moment to review :)

jstebbins commented on gtk/src/queuehandler.c:
```diff
@@ -1456,26 +1456,11 @@ ghb_update_all_status(signal_user_data_t *ud, int status)
+    dialog = gtk_file_chooser_dialog_new("Export Queue",
                       hb_window,
                       GTK_FILE_CHOOSER_ACTION_SAVE,
                       GHB_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
```
First off, thanks!

PR looks mostly good.  But does not build :grimacing: 

There is a use of variable ```queue``` just below this point ```ghb_value_free(&queue);``` before queue is defined.

soulshadow replied:
ahh I see my mistake.

i'm a bit new to github collaboration, whats the correct way to submit a corrected version?

soulshadow replied:
ok, lets see if I got that right

sr55 replied:
yup, you just commit the next change and push to your fork. The Pull request should update automatically.

soulshadow committed:
Update queuehandler.c

refactored the if-block handling the gtk dialog response to work with the first change

soulshadow commented:
so, looking over the build messages one thing jumps out at me. In my inexperience with large projects I neglected to consider the impact on translations that might arise due to the change to the file dialog window names. I'm not sure if this will be an issue or how big of an issue it could be?

sr55 commented:
Looks like your latest commit builds. 

@jstebbins regarding translations ^?

jstebbins commented:
> so, looking over the build messages one thing jumps out at me. In my inexperience with large projects I neglected to consider the impact on translations that might arise due to the change to the file dialog window names. I'm not sure if this will be an issue or how big of an issue it could be?

Your string changes are an improvement. And actually, those particular strings were not currently being properly translated. To tag strings for translation, they need to be enclosed in the ```_()``` macro.  So e.g. ```_("Export Queue")```.

So if you wouldn't mind adding the macro, the rest looks good and I'll push.

soulshadow committed:
add translation macro

add translation macro to gtk dialog window's names that were changed in previous commits.

soulshadow commented:
Macro added to the 2 strings.  A quick glance through the surrounding code shows additional strings, mainly gtk labels, that might benefit from translation as well. However, probably best to leave that for it's own separate commit.

jstebbins commented:
> Macro added to the 2 strings. A quick glance through the surrounding code shows additional strings, mainly gtk labels, that might benefit from translation as well. However, probably best to leave that for it's own separate commit.

Yeah, I set up the gtk project to be translatable when I first started it (2008), but I had no translators so was not very diligent about marking all strings that need translation. Most are actually marked, but there are still many that are not.

A separate PR for this would be very much appreciated :grin:

jstebbins commented:
One more thing I'd like to point out.  When making corrections to a previous commit in a PR, it is helpful to the reviewers if you squash the original commit and the fix commit together. You can do this easily with git's interactive rebase command. E.g. ```git rebase -i HEAD^^^``` will allow you to modify the last 3 commits. With that command you can reorder commits and mark commits as "fixes" to the preceding commit to squash them.  Then you can force push to your PR branch so that reviewrers have a cleaner version of the changes to look at.

soulshadow commented:
thanks for the tip.  I'll try to remember that for the future.  In this instance, mostly just a little copy and paste, I did everything directly through github's interface, just figuring it out as I went along.  I always appreciate the chance to learn something new.
